types/span.go

```diff
@@ -14,7 +13,6 @@ import (
 //   values, respectively.
 type Span struct {
 	CoreSpanMetadata
-	Annotations       []*Annotation          `json:"annotations,omitempty"` // TODO lift annotation struct definition into this file
```

Comment:
This didn't seem to be used? At least no tests are failing :eyes:

Reply:
Please leave it for now so we can limit the changes in this PR.

Reply:
Ok, I added it back. But _before_ it used the same `Annotation` as `types.ZipkinJSONSpan` (now `types.v1.ZipkinJSONSpan`). I created a new `annotation` struct restricted to the Span object to avoid interlocked dependencies between Span and the zipkin objects.

If you want me to limit the scope of changes even further I can go back to using the same `Annotation` struct that the v1 `ZipkinJSONSpan` uses, but I'll need to revert the subpackaging `types.v1` and `types.v2` that I came up with, to avoid a circular dependency.